test/BEST_PRACTICES.md

```diff
@@ -0,0 +1,54 @@
+# Test Best Practices
+
+This is a collection of best practices to write high quality tests. The best practices
+can be referenced in review comments to speed up the review process.
+
+- [Go Style](#go-style)
+- [Test Case Traceability](#test-case-traceability)
+- [Don't Sleep](#dont-sleep)
+- [Independent](#independent)
+- [Secrets](#secrets)
+- [Logging](#logging)
+
+## Go Style
+
+We don't want to redefine best practices for go, therefore we should reuse the existing ones
+described in the [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
+and in the[Effective Go](https://golang.org/doc/effective_go.html) documents.
+
+## Test Case Traceability
+
+In order to compute test coverage and test automation progress across multiple test suites all
+automated tests must have an ID and must be traced back to the [integreatly-test-cases](https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases)
+repo following the [How to automate a test case and link it back](https://gitlab.cee.redhat.com/integreatly-qe/integreatly-test-cases#how-to-automate-a-test-case-and-link-it-back)
+tutorial.
+
+## Don't Sleep
+
+When waiting for a resource to become available or an operation to complete do not use fixed wait time,
+instead you should create a loop and keep pooling until the operation doesn't complete.
+
+## Independent
+
+A test must always be independent of other tests, which mean:
+
+- it should not rely on other tests changes so that it would be possible to run it independently
```

Comment:
```suggestion
- it should not rely on the successful execution of other tests so the test can be run independently
```

Reply:
@damienomurchu I don't get why removing the `so that`, I would like to emphasize that there is a reason why we do it and a consequence if you don't do it and I think that using `so that` will strength the sentence. I would like just o understand why removing it?